api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -182,6 +183,9 @@ message Permission {
     // A port number that describes the destination port connecting to.
     uint32 destination_port = 6 [(validate.rules).uint32 = {lte: 65535}];
 
+    // A port number range that describes a range of destination ports connecting to.
+    type.v3.Int32Range destination_port_range = 11;
```

Comment:
A few questions:
1. Are there any constraints on the values here? Is there a default value if it is not specified?
2. How does this interact with `destination_port` above? Should the previous field be deprecated?
3. Could there be a use-case of multiple ranges (should this be a repeated field)?

Reply:
1. The `destination_port_range` in RBAC is the same with `destination_port_range` in listener filter chain match predicate. Maybe it will be natural to add a constraint where the port must between 0 and 65535, and the `start` port must be less than the `end` port, but an input like [443, 443), [443, 80) or [-1, 65537) will not actually cause any exceptions, the matcher still works as usual. And Also, there is no default value.

https://github.com/envoyproxy/envoy/blob/9d0f0355135a19dff6949fed884e81b0d1eb101a/api/envoy/config/listener/v3/listener_components.proto#L330-L332

2. The `destination_port` and `destination_port_range` can work together using logic rule `or_rules` in RBAC. For example, a rule set like

   ```yaml
   - or_rules:
       rules:
         - destination_port_range:
             start: 21
             end: 23
         - destination_port: 80
   ```

   will filter out connections except there destination port is 21, 22 or 80. I am not sure whether the `destination_port` should be deprecated or not, since I am new to Envoy, and not aware of users' request.

3. Yes. A rule set like

   ```yaml
   - or_rules:
       rules:
         - destination_port_range:
             start: 21
             end: 23
         - destination_port_range:
             start: 30000
             end: 40000
   ```

   is valid.

Reply:
@htuch for another set of eyes on this:
1. Should `destination_port_range` deprecate the `destination_port`?
2. Do you think making the new field repeated makes sense?

Reply:
1. If we were designing from scratch, I might be tempted to go there, but given the material UX simplification for a common case and the churn of deprecation, I think it's fine to say what is here is fine. Range matches are likely to be rarer than singleton (or we would have seen this PR by now).
2. Don't feel super strongly either way, if it's not going to be very common to be punching lots of holes in ranges then a single int range with RBAC conjunction/disjunction rules is going to be fine.

Reply:
1. In fact, I also agree with @htuch that the `destination_port_range` should not be deprecate `destination_port` in consideration of compatibility. The purpose of `destination_port_range` is introduced by #16039 which using large amount of ports for dynamic forward proxy, and not all the scene want to replace the single-line `destination_port` with the 3-line `destination_port_range`.
2. `destination_port_range` itself is not a repeated field, but all the rule in RBAC permission can be repeated with the `or_rules`.

Reply:
I see, thanks for explaining this.